bin-ranges-transfer/src/test/java/uk/gov/pay/java_lambdas/bin_ranges_transfer/AppTest.java

```diff
@@ -38,57 +39,64 @@
             Path.of(ClassLoader.getSystemResource("ssh/test_private_key.rsa").getPath()))
         );
         testPassphrase = String.join("", Files.readAllLines(
             Path.of(ClassLoader.getSystemResource("ssh/test_passphrase").getPath())
         ));
-        mockSftpServer = new MockSftpServer();
-        mockSftpServer.startServer();
     }
 
     @AfterEach
     public void tearDown() throws Exception {
         mockSftpServer.stopServer();
+        constantsMockedStatic.close();
+        dependencyFactoryMockedStatic.close();
     }
 
     @Test
-    void handleRequest_shouldSuccessValue() throws GeneralSecurityException, IOException {
-        S3Client mockS3Client = mock(S3Client.class);
-        SsmClient mockSsmClient = mock(SsmClient.class);
-        
-        try (MockedStatic<DependencyFactory> dependencyFactoryMockedStatic = mockStatic(DependencyFactory.class, CALLS_REAL_METHODS);
-             MockedStatic<Constants> constantsMockedStatic = mockStatic(Constants.class)) {
-            
-            dependencyFactoryMockedStatic.when(DependencyFactory::s3Client).thenReturn(mockS3Client);
-            dependencyFactoryMockedStatic.when(DependencyFactory::ssmClient).thenReturn(mockSsmClient);
-            constantsMockedStatic.when(Constants::getSftpPort).thenReturn(mockSftpServer.getPort());
-            constantsMockedStatic.when(Constants::getSftpHost).thenReturn(mockSftpServer.getHost());
-            constantsMockedStatic.when(Constants::getSftpUsername).thenReturn(TEST_SERVER_USERNAME);
-
-            when(mockSsmClient.getParameter(any(GetParameterRequest.class)))
-                .thenReturn(ssmParameterResponseBuilder(testPassphrase))
-                .thenReturn(ssmParameterResponseBuilder(testPrivateKey));
+    void handleRequest_shouldReturnCandidate() throws GeneralSecurityException, IOException {
+        when(mockSsmClient.getParameter(any(GetParameterRequest.class)))
+            .thenReturn(ssmParameterResponseBuilder(testPassphrase))
+            .thenReturn(ssmParameterResponseBuilder(testPrivateKey));
 
-            App function = new App();
-            String result = function.handleRequest(null, context);
-            assertEquals("Success: downloaded BIN ranges from Worldpay", result);
+        App function = new App();
+        Candidate result = function.handleRequest(null, mockContext);
+        assertTrue(result.s3Key().contains(V03_FILENAME));
+        assertTrue(result.proceed());
 
-            verify(mockS3Client, atMost(2)).putObject(any(PutObjectRequest.class), any(RequestBody.class));
-            dependencyFactoryMockedStatic.verify(DependencyFactory::s3Client);
-            dependencyFactoryMockedStatic.verify(DependencyFactory::ssmClient);
-            dependencyFactoryMockedStatic.verify(() -> DependencyFactory.sshClient(any(Path.class), eq(testPassphrase)));
-        }
+        verify(mockS3Client, atMost(2)).putObject(any(PutObjectRequest.class), any(RequestBody.class));
```

Comment:
Can we really not match more precisely on the `PutObjectRequest` and `RequestBody`?

Reply:
yes we could be more precise, for this test I'm more interested in knowing that the request was made though